main.py

```diff
@@ -467,8 +467,11 @@ def main(cl_arguments):
         tb_logdir = os.path.join(args.run_dir, "tensorboard")
         _run_background_tensorboard(tb_logdir, cl_args.tensorboard_port)
 
+    if args.do_full_eval:
+        # We evaluate on all target tasks during evaluation stage.
+        eval_tasks = copy.deepcopy(target_tasks)
```

Comment:
I think it'd be easier to maintain if we did this entirely inside target train. Evaluate on all the target tasks, but create a copy later on with something like `target_tasks_to_train =  copy.deepcopy(target_tasks)`, and only pass those to the trainer.

This is partially just because it's not clear why these two should be different from their names as is.

Reply:
Sounds good